### PR TITLE
Add .NET CoreCLR build script

### PR DESCRIPTION
**Pull request by hez2010**
This PR continues #7.

### CODE REVIEW

**Comment by partouf** [comment]
merging to test

File: build/build-dotnet.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+set -ex
+
+VERSION=$1
+if echo ${VERSION} | grep 'trunk'; then
+    BRANCH=main
+else
+    BRANCH=release/${VERSION}
```
Comment (partouf) [inline build/build-dotnet.sh:9]: The `release/6.0` branch seems to be in active development while the official release is tagged as `v6.0.0` - it's better to use the tags here and only have trunk/main as an active development version.

File: build/build-dotnet.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+set -ex
+
+VERSION=$1
+if echo ${VERSION} | grep 'trunk'; then
+    BRANCH=main
```
Comment (kunalspathak) [inline build/build-dotnet.sh:7]: How frequently are these builds produced?

Comment (partouf) [reply build/build-dotnet.sh:7]: nightlies are not produced yet, we'll need to work on making the process of build to live is shorter, especially the nuget packaging which we somehow need to automate together with new versions. For now, let's see what users like/dislike/want/etc.

**Pull request merged**